### ISSUE TITLE
Error: An item with the same key has already been added.

### ISSUE THREAD

**Issue by 1N50MN14** [Azure/azure-webjobs-sdk#1368]
Please provide a succinct description of the issue.

#### Repro steps

Provide the steps required to reproduce the problem

1. Step A
I had a routing issue with proxies, even though I was explicitly specifying a backend uri to match a specific slot, it the proxy was forwarding requests to a wrong, random slot. After spending 3 hours on this I finally figured out `ROUTING_EXTENSION_VERSION` needed to be changed to `~0.3`. 

But now I'm receiving HTTP error 400 on my API calls and the portal is throwing the above error. 

1. It would be nice if the error could specify what, where, which key is it referring to; I had to first figure out it was Application Settings, then delete ALL of the keys one by one to find out which key was causing the issue.

2. Now I can only delete this property for the error to go away... and pray routing will work.

2. Step B

Provide a description of the expected behavior.
More explicit error, proxies to route to the correct slot destination.

#### Actual behavior
Unhelpful error message.

#### Known workarounds
Still trying to figure it out.

**Comment by 1N50MN14** [comment]
Ok, **workaround**:
Individually in each slot delete `ROUTING_EXTENSION_VERSION`, refresh, re-add with value ~0.2

**Comment by paulbatum** [comment]
@safihamid I thought i saw something related to this... do you already have a fix?

**Comment by safihamid** [comment]
I am trying to repro your issue and see if it has already been fixed or not. could you please make sure your function app runtime version is equal or higher than 1.0.11296.0 and then you can delete all your ROUTING_EXTENSION_VERSION app settings and see if you still get that error?

If you do, could you please share your proxies.json file and a simple repro scenario?
Thanks

**Comment by 1N50MN14** [comment]
@safihamid This issue was fixed, workaround in my previous comment.

A more verbose error message is still needed in case others encounter this in the future..

**Comment by safihamid** [comment]
@1N50MN14 the new function runtime versions  equal or above 1.0.11296.0 do not requires that setting which you have in your workaround and in fact it gets ignored. if you still see that issue with the new runtime please do let us know as the workaround will not work.

**Comment by safihamid** [comment]
Also make sure you don't have a function name and a proxy name with the same exact name as that would cause this issue. I just filed this bug: https://github.com/Azure/azure-functions-ux/issues/1878

**Comment by safihamid** [comment]
And I sent a pull request to fix this in function runtime. basically you will get this error if there is a proxy and function with the same name in the same function app. we will return a more verbose error message when this happens. thanks for reporting this issue.

**Comment by 1N50MN14** [comment]
@safihamid oh I didn't notice that, if you look at my proxy for that specific function app, I do have a proxy name `updateProfile' (my proxies are auto-generated using a local script) which points to a funciton app with the same name, as of now it does work just fine.

When it comes to the new runtime version is that already rolled out? because of so then you're using I can simply delete this setting from all slots? This setting was automatically generated when I created the slots and the issue was caused due a mismatch in its value as I didn't create the slots at the same time..

**Comment by 1N50MN14** [comment]
Either cases I will add a sanity check to make sure there are no conflicting function and proxy names, thanks for letting me know

**Comment by safihamid** [comment]
Yes the new runtime is already rolled out but won't have this fix as I just fixed this today and will go out with the next release. you will get the new runtime loaded when you restart your function app or when app service releases an update and restarts your app. for now just make sure there are no duplicate names between functions and proxies and everything should work fine.

**Comment by 1N50MN14** [comment]
@safihamid awesome thank you!
